Include/cpython/compile.h

```diff
@@ -31,11 +31,26 @@ typedef struct {
 #define _PyCompilerFlags_INIT \
     (PyCompilerFlags){.cf_flags = 0, .cf_feature_version = PY_MINOR_VERSION}
 
+/* source location information */
+typedef struct {
+    int lineno;
+    int end_lineno;
+    int col_offset;
+    int end_col_offset;
+} PyCompilerSrcLocation;
```

Comment:
I wonder if I should put an underscore so it's ``_PyCompilerSrcLocation``. Are we ready to make this public?